src/Task.cc

```diff
@@ -3701,6 +3701,57 @@ static void copy_mem_mapping(Task* from, Task* to, const KernelMapping& km) {
   }
 }
 
+static void copy_mem_mapping_just_used(Task* from, Task* to, const KernelMapping& km)
+{
+  ScopedFd& fd = from->vm()->pagemap_fd();
+  if (!fd.is_open()) {
+    FATAL() << "Failed to open " << from->proc_pagemap_path();
+  }
+
+  size_t pagesize = page_size();
+  void* p = (void*)km.start().as_int();
+  const auto MAP_SIZE = km.size();
+#if defined(__i386__)
+  const auto BUF_PAGE_COUNT = 64*1024;
+#else
+  const auto BUF_PAGE_COUNT = 1024*1024;
+#endif
+
+  uint64_t* buf = new uint64_t[BUF_PAGE_COUNT];
+  uint64_t pages_present = 0;
+  for (uintptr_t offset = 0; offset < MAP_SIZE; offset += pagesize*BUF_PAGE_COUNT) {
+    pread(fd, buf, BUF_PAGE_COUNT*sizeof(uint64_t), 8*(((uintptr_t)p + offset)/pagesize));
+    for (size_t i = 0; i < BUF_PAGE_COUNT; ++i) {
+      if (buf[i] & (1LL << 63)) {
```

Comment:
Document what this is checking.

Reply:
I added a comment and tried to get a better readability in V2.